mlflow/transformers/__init__.py

```diff
@@ -1363,6 +1363,28 @@ def _build_pipeline_from_model_input(model_dict: Dict[str, Any], task: Optional[
         ) from e
 
 
+def _get_task_for_model(model_name_or_path: str, default_task=None) -> str:
+    """
+    Get the Transformers pipeline task type fro the model instance.
+
+    NB: The get_task() function only works for remote models available in the Hugging
+    Face hub, so the default task should be supplied when using a custom local model.
+    """
+    from transformers.pipelines import get_task
+
+    try:
+        return get_task(model_name_or_path)
+    except RuntimeError as e:
```

Comment:
Shall we check the exception message ?  if the exception is "Repo id must be in the form..." then we execute the following logic, otherwise we still raise the error.

Reply:
We could, but it may be brittle as the error message can change. I think even if the reason of failing is not repo id, the error message below is still correct?